packages/material-ui/src/internal/SelectableGroup/useSelectedState.js

```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import SelectableGroupContext from './SelectableGroupContext';
+
+/**
+ * @ignore - internal component.
+ */
+function useSelectedState() {
```

Comment:
We can use `React.useContext(SelectableGroupContext)` directly in the components. Why abstracting it, are we doing it too soon?

Reply:
It's not abstraction it's readability while hiding implementation details. Just because we can doesn't mean we should.

Reply:
readability, I haven't noticed the difference. My point of interest is the potential module overhead removal by overhead, I mean from a maintainer perspective: "we have x dependencies, what each dependency is doing? vs I have x - 1."
Now, I have a co-worker that loves creating modules for 1 line functions. I understand the motivation.